### PR TITLE
Convert dataclasses to pydantic

### PR DESCRIPTION
**Pull request by petern-annotell**
All remaining `@dataclass`-es have been ported over to pydantic models.

Quite a lot of changes that fall into some patterns:
- Replacing the dataclass annotations by extending from `BaseSerializer`
- Removing custom `to_dict` and `from_json` impls and adding field aliases instead
- Set default values on fields that previously could be empty, so that they don't have to be specified everywhere
- Juggled a lot of imports with @olinoskar 's help; this prevented a circular dependency but means some classes that could be imported from any input model sub-package no longer can.

Clients will need to update their code. I will create a separate PR for updating the docs with a migration guide.

### CODE REVIEW

File: kognic-io/kognic/io/model/projects/project_batch.py [hunk 1]
```diff
@@ -15,22 +13,12 @@ class ProjectBatchStatus(str, Enum):
...
-    def from_json(js: dict):
-        return ProjectBatch(
-            project=js["project"],
-            batch=js["batch"],
-            title=js["title"],
-            status=ProjectBatchStatus(js["status"]),
-            created=ts_to_dt(js["created"]),
-            updated=ts_to_dt(js["updated"])
-        )
+    # TODO for created/updated does ts_to_dt need to be given as a validator (or something) with pydantic,
```
Comment (kallus) [inline kognic-io/kognic/io/model/projects/project_batch.py:24]:
```suggestion
```
I think we should avoid checking in TODO lines. I suggest adding this to jira instead if it in not convenient to solve it now.

**Review by kallus — CHANGES_REQUESTED** [review]
Very nice to see a negative change of LOC count! :)

File: kognic-io/kognic/io/model/projects/project_batch.py (hunk 1 above)
Comment (petern-annotell) [reply kognic-io/kognic/io/model/projects/project_batch.py:24]: Yep good catch :+1: will nuke it

**Review by kallus — APPROVED** [review]

**Review by kallus — APPROVED** [review]
LGTM

**Pull request merged**